### ISSUE TITLE
run.sh failed without error message

### ISSUE THREAD

**Issue by ltalirz** [aiidalab/aiidalab-docker-stack#194]
I tried running the following on MacOS 11.5.2 with Docker Desktop 3.6.0
``` 
./run.sh --no-build --image aiidalab/aiidalab-docker-stack:latest 8888 /Users/leopold/Personal/Postdoc-MARVEL/repos/aiidalab/aiidalab-docker-stack/vol
Launching the container...
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Waiting for container to start...
``` 
The script returned without giving me a link to a container or an error message.

The container did not start; I don't know why

``` 
$ docker logs 4bbe7e27a9bd
*** Running /etc/my_init.d/00_regen_ssh_host_keys.sh...
*** Running /etc/my_init.d/10_create-system-user.sh...
/usr/bin/id: ‘aiida’: no such user
no change     /opt/conda/condabin/conda
no change     /opt/conda/bin/conda
no change     /opt/conda/bin/conda-env
no change     /opt/conda/bin/activate
no change     /opt/conda/bin/deactivate
no change     /opt/conda/etc/profile.d/conda.sh
no change     /opt/conda/etc/fish/conf.d/conda.fish
no change     /opt/conda/shell/condabin/Conda.psm1
no change     /opt/conda/shell/condabin/conda-hook.ps1
no change     /opt/conda/lib/python3.7/site-packages/xontrib/conda.xsh
no change     /opt/conda/etc/profile.d/conda.csh
no change     /home/aiida/.bashrc
No action taken.
*** Running /etc/my_init.d/10_syslog-ng.init...
Sep 13 10:37:25 4bbe7e27a9bd syslog-ng[268]: syslog-ng starting up; version='3.13.2'
*** Running /etc/my_init.d/20_start-rabbitmq.sh...
 * Starting RabbitMQ Messaging Server rabbitmq-server
   ...done.
*** Running /etc/my_init.d/30_start-postgres.sh...
pg_ctl: no server running
waiting for server to start.... stopped waiting
pg_ctl: could not start server
Examine the log output.
*** /etc/my_init.d/30_start-postgres.sh failed with status 1

*** Killing all processes...
Sep 13 10:37:43 4bbe7e27a9bd syslog-ng[268]: syslog-ng shutting down; version='3.13.2'
``` 

@yakutovicha any hints?

**Comment by ltalirz** [comment]
I now repeated the instructions with rebuilding the docker image locally via
```
./run.sh 8888 /Users/leopold/Personal/Postdoc-MARVEL/repos/aiidalab/aiidalab-docker-stack/vol
``` 

and got the same error

**Comment by ltalirz** [comment]
Writing into the mounted folder started already
```
 ~/Personal/Postdoc-MARVEL/repos/aiidalab/aiidalab-docker-stack/vol [develop|⚑ 2]
17:01 $ ls -ltra
total 16
-rw-r--r--  1 leopold staff  220 Sep 13 12:34 .bash_logout
-rw-r--r--  1 leopold staff  807 Sep 13 12:34 .profile
drwx------  5 leopold staff  160 Sep 13 12:35 .ssh
-rw-r--r--  1 leopold staff 4449 Sep 13 12:35 .bashrc
-rw-r--r--  1 leopold staff    0 Sep 13 12:35 test
drwxr-xr-x  9 leopold staff  288 Sep 13 12:35 .
drwxr-xr-x  5 leopold staff  160 Sep 13 15:47 .rabbitmq
drwxr-xr-x 20 leopold staff  640 Sep 13 16:58 ..
drwx------ 25 leopold staff  800 Sep 13 17:00 .postgresql
```

Including the postgresql log file
```
2021-09-13 13:47:18.007 UTC [1633] LOG:  listening on IPv4 address "127.0.0.1", port 5432
2021-09-13 13:47:18.009 UTC [1633] LOG:  could not bind IPv6 address "::1": Cannot assign requested address
2021-09-13 13:47:18.009 UTC [1633] HINT:  Is another postmaster already running on port 5432? If not, wait a few seconds and retry.
2021-09-13 13:47:18.018 UTC [1633] LOG:  could not bind Unix address "/home/aiida/.postgresql/.s.PGSQL.5432": Invalid argument
2021-09-13 13:47:18.018 UTC [1633] HINT:  Is another postmaster already running on port 5432? If not, remove socket file "/home/aiida/.postgresql/.s.PGSQL.5432" and retry.
2021-09-13 13:47:18.019 UTC [1633] WARNING:  could not create Unix-domain socket in directory "/home/aiida/.postgresql"
2021-09-13 13:47:18.020 UTC [1633] FATAL:  could not create any Unix-domain sockets
2021-09-13 13:47:18.023 UTC [1633] LOG:  database system is shut down
```

**Comment by ltalirz** [comment]
Adding
```
listen_addresses = '*'
```
to `.postrgresql/postgresql.conf` gets rid of the first error but leaves the second one:
``` 
2021-09-13 15:39:33.138 UTC [1631] LOG:  listening on IPv4 address "0.0.0.0", port 5432
2021-09-13 15:39:33.140 UTC [1631] LOG:  listening on IPv6 address "::", port 5432
2021-09-13 15:39:33.149 UTC [1631] LOG:  could not bind Unix address "/home/aiida/.postgresql/.s.PGSQL.5432": Invalid argument
2021-09-13 15:39:33.149 UTC [1631] HINT:  Is another postmaster already running on port 5432? If not, remove socket file "/home/aiida/.postgresql/.s.PGSQL.5432" and retry.
2021-09-13 15:39:33.150 UTC [1631] WARNING:  could not create Unix-domain socket in directory "/home/aiida/.postgresql"
2021-09-13 15:39:33.150 UTC [1631] FATAL:  could not create any Unix-domain sockets
2021-09-13 15:39:33.154 UTC [1631] LOG:  database system is shut down
``` 

On second look, this error message is indeed a bit weird.
What is this `/home/aiida/.postgresql/.s.PGSQL.5432` file?

**Comment by ltalirz** [comment]
Interestingly, following the [instructions for the aiida-core 1.6.4 container](https://aiida.readthedocs.io/projects/aiida-core/en/latest/intro/run_docker.html) works just fine and the unix-domain socket file is there:

```
srwxrwxrwx  1 aiida aiida     0 Sep 13 17:07 .s.PGSQL.5432
```

However, this container does not bind-mount another volume inside the home folder.
I start to suspect that the problem is that something about my setup (using an x86 container on an arm64 chip on macos) prevents the socket file from living in the mounted directory.

Of course, the socket file is really a temporary thing so we could simply decide to place it somewhere else.
Let me try that.

**Comment by ltalirz** [comment]
Fixed in https://github.com/aiidateam/aiida-prerequisites/pull/32

**Comment by ltalirz** [comment]
fixed in 3977b7632775358a7239e146608921e1ce41d863
